plugins/homepage/web_client/views/ConfigView.js

```diff
@@ -17,7 +17,54 @@ var ConfigView = View.extend({
             this._saveSettings([{
                 key: 'homepage.markdown',
                 value: this.editor.val()
+            },
+            {
+                key: 'homepage.header',
+                value: this.$('#g-homepage-header').val()
+            },
+            {
+                key: 'homepage.subheader',
+                value: this.$('#g-homepage-subheader').val()
+            },
+            {
+                key: 'homepage.welcome_text',
+                value: this.welcomeText.val()
+            },
+            {
+                key: 'homepage.logo',
+                value: this.logoUrl
             }]);
+        },
+
+        'change #g-homepage-logo': function (event) {
+            var reader = new FileReader();
```

Comment:
Encoding this as a data URI is more complicated, will require the entire image to be stored in the database as a setting, and will require the entire image to be re-downloaded inside the REST endpoint every time the homepage is loaded. Instead, we should make this be an ordinary Girder file upload, which will upload its content into the existing `Homepage Assets` collection. The setting can then store the `fileId` of the uploaded image, and use it to construct a local URL to embed the image (which the browser can cache).

Reply:
I don't know if this is relevant, but for QIDW (just before work on this PR began), I did something very similar where the logo was an item uploaded to girder, and then the fileId was the setting.  But I also took URI construction out of the equation for clients by moving that logic over to a dedicated server-side route that would just look up the setting, and return a redirect to the right file.  This way, clients can point to a specific URI and know that they'll end up fetching the right file, even after changes to the settings.